app/assets/javascripts/admin/analytics-modules/ga4-index-section-setup.js

```diff
@@ -12,10 +12,21 @@ window.GOVUK.analyticsGa4.analyticsModules =
 
       moduleElements.forEach(function (moduleElement) {
         const indexedElements = moduleElement.querySelectorAll(
-          'select, input:not([data-module~="select-with-search"] input)'
+          'select, input:not([data-module~="select-with-search"] input):not([type="radio"]):not([type="hidden"]), fieldset'
         )
+
         indexedElements.forEach((element, index) => {
-          element.dataset.ga4IndexSection = index
+          if (element.tagName === 'FIELDSET' || !element.closest('fieldset')) {
+            const indexData = {
+              // cast to string otherwise treated as false by
+              // `analyticsGa4.core.applySchemaAndSendData`
+              index_section: String(index),
+              index_section_count: String(indexedElements.length)
```

Comment:
Is that comment true? As best I can tell, we're only doing [explicit checks for `undefined` and `null`](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-schemas.js#L58-L64) in the corresponding code. Feels as though any casting should be done in the core module rather than in the consumer.

Reply:
Interesting, I remember this behaviour happening but actually when I try it now it doesn't seem to do that. It might have been me getting confused with something else or another index property overwriting it. Can remove this casting for sure.